tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java

```diff
@@ -358,6 +358,29 @@ public void testSymbolLookupInFields(int line, int column, int expSymbols, List<
         };
     }
 
+    @Test
+    public void testOnFailClauseSymbolLookup() {
+        Project project = BCompileUtil.loadProject("test-src/on_fail_symbol_lookup_test.bal");
+        Package currentPackage = project.currentPackage();
+        ModuleId defaultModuleId = currentPackage.getDefaultModule().moduleId();
+        PackageCompilation packageCompilation = currentPackage.getCompilation();
+        SemanticModel model = packageCompilation.getSemanticModel(defaultModuleId);
+        Document srcFile = getDocumentForSingleSource(project);
+
+        BLangPackage pkg = packageCompilation.defaultModuleBLangPackage();
+        ModuleID moduleID = new BallerinaModuleID(pkg.packageID);
+
+        List<Symbol> allInScopeSymbols = model.visibleSymbols(srcFile, LinePosition.from(25, 23));
+        List<Symbol> symbols = new ArrayList<>();
+        for (Symbol symbol : allInScopeSymbols) {
+            if (symbol.getModule().get().id().equals(moduleID) && symbol.kind() == SymbolKind.VARIABLE) {
+                symbols.add(symbol);
+            }
+        }
```

Comment:
`SemanticAPITestUtils` has a function `getSymbolsInFile()`. We can get the symbols filtered. shall we use this?

Reply:
Updated in d8a85d3cf66bd59ad3827b8092e1dbe7c59bb0d7